File: backend/H4H.Core/Models/Review.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace H4H.Core.Models
{
    [Table("reviews")]
    public class Review
    {
        [Column("id")]
        public Guid Id { get; set; }

        [Column("appointment_id")]
        public Guid AppointmentId { get; set; }

        [Column("client_id")]
        public Guid ClientId { get; set; }

        [Column("specialist_id")]
        public Guid SpecialistId { get; set; }

        [Column("rating")]
        public int Rating { get; set; }

        [Column("comment")]
        public string? Comment { get; set; }

        [Column("is_verified")]
        public bool IsVerified { get; set; } = true;

        [Column("created_at", TypeName = "timestamp without time zone")]
        public DateTime CreatedAt { get; set; } = DateTime.SpecifyKind(DateTime.Now, DateTimeKind.Unspecified);

        [Column("updated_at", TypeName = "timestamp without time zone")]
        public DateTime UpdatedAt { get; set; } = DateTime.SpecifyKind(DateTime.Now, DateTimeKind.Unspecified);

        public virtual Appointment Appointment { get; set; } = null!;
        public virtual Client Client { get; set; } = null!;
        public virtual Specialist Specialist { get; set; } = null!;
    }
}

File: marketplace_app/lib/models/review.dart
class Review {
  final String id;
  final String appointmentId;
  final String clientId;
  final String specialistId;
  final int rating;
  final String? comment;
  final bool isVerified;
  final DateTime createdAt;
  final DateTime updatedAt;

  Review({
    required this.id,
    required this.appointmentId,
    required this.clientId,
    required this.specialistId,
    required this.rating,
    this.comment,
    required this.isVerified,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['id'],
      appointmentId: json['appointmentId'],
      clientId: json['clientId'],
      specialistId: json['specialistId'],
      rating: json['rating'],
      comment: json['comment'],
      isVerified: json['isVerified'],
      createdAt: DateTime.parse(json['createdAt']),
      updatedAt: DateTime.parse(json['updatedAt']),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'appointmentId': appointmentId,
      'clientId': clientId,
      'specialistId': specialistId,
      'rating': rating,
      'comment': comment,
      'isVerified': isVerified,
      'createdAt': createdAt.toIso8601String(),
      'updatedAt': updatedAt.toIso8601String(),
    };
  }

  Rating toRating() {
    return Rating(rating: rating, comment: comment ?? '');
  }
}

class Rating {
  final int rating;
  final String? comment;

  Rating({required this.rating, this.comment});

  factory Rating.fromJson(Map<String, dynamic> json) {
    return Rating(rating: json['rating'], comment: json['comment']);
  }

  Map<String, dynamic> toJson() => {'rating': rating, 'comment': comment};
}
